tf_agents/agents/dp/dp_agent.py

```diff
@@ -0,0 +1,214 @@
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+
+from tf_agents.environments import py_environment
+from tf_agents.typing import types
+
+class DpAgent(object):
```

Comment:
Inherit from the TFAgent Base class: https://github.com/tensorflow/agents/blob/cdede4517bbb182f8a5519ba57fb781044164711/tf_agents/agents/tf_agent.py#L43